File: api/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class Customer
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(30)]
        public string Names { get; set; } = default!;
        [Required]
        [MaxLength(30)]
        public string Surnames { get; set; } = default!;
        [Required]
        [MaxLength(15)]
        public string Dni { get; set; } = default!;
        [MaxLength(150)]
        public string? Address { get; set; } = default!;
        [Required]
        [MaxLength(15)]
        public string Phone { get; set; } = default!;
        [Required]
        [MaxLength(50)]
        public string Email { get; set; } = default!;
        public ICollection<Sale> Sales { get; set; } = new List<Sale>();
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

    }
}

File: api/app/models/customer.ts
import { DateTime } from 'luxon';
import { BaseModel, column, hasMany } from '@adonisjs/lucid/orm';
import Sale from './sale.js';
import type { HasMany } from '@adonisjs/lucid/types/relations';

export default class Customer extends BaseModel {
  @column({ isPrimary: true })
  declare id: number;

  @column()
  declare names: string;

  @column()
  declare surnames: string;

  @column()
  declare address: string | null;

  @column()
  declare phone: string;

  @column()
  declare email: string;

  @column()
  declare dni: string;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @hasMany(() => Sale)
  declare sales: HasMany<typeof Sale>;
}
